study-app/src/components/Main/Main.jsx

```diff
@@ -1,61 +1,80 @@
 import { useState, useEffect } from "react";
-//import FrontEndRoadmap from "../FrontEndRoadmap/FrontEndRoadmap";
-//import BackEndRoadmap from "../BackEndRoadmap/BackEndRoadmap";
 import LeftSideMenu from "../LeftSideMenu/LeftSideMenu";
 import Welcome from "../Welcome/Welcome";
 import RightSideMenu from "../RightSideMenu/RightSideMenu";
 import axios from "axios";
 import RoadMap from "../RoadMap/RoadMap";
 
-
 function Main() {
+  const [courses, setCourses] = useState([]);
+  const [mode, setMode] = useState("");
+  const [nodes, setNodes] = useState([]);
+  const [edges, setEdges] = useState([]);
+  const [courseName, setCourseName] = useState("");
+  const [lessonData, setLessonData] = useState([]);
 
-	const [courses, setCourses] = useState([]);
-	const [mode, setMode] = useState("");
-	const [nodes, setNodes] = useState([]);
-	const [edges, setEdges] = useState([]);
-	const [courseName, setCourseName] = useState("")
-
-	const [selectedNode, setSelectedNode] = useState("");
-
-	const allCourses = async() => {
-		try{
-			const response = await axios.get("http://localhost:3004/course")
-			console.log(response.data);
-			setCourses(response.data)
-		} catch(err) {
-			console.log(err)
-		}
-	}
-	
-	useEffect(() => {
-		allCourses()
-	}, []);
-
+  const [selectedNode, setSelectedNode] = useState("");
 
+  const allCourses = async () => {
+    try {
+      const response = await axios.get("http://localhost:3004/course");
+      console.log(response.data);
+      setCourses(response.data);
+    } catch (err) {
+      console.log(err);
+    }
+  };
 
-	// const modes = {
-	// 	"Front End": <FrontEndRoadmap setSelectedNode={setSelectedNode} />,
-	// 	"Back End": <BackEndRoadmap setSelectedNode={setSelectedNode} />,
-	// };
+  const getLessonDetails = async (e) => {
+    try {
+      const id = e.target.id;
+      if (!id) return;
+      const lessonDetails = await axios.get(
+        `http://localhost:3004/lesson/${id}`
+      );
+      console.log("lessonDetails", lessonDetails.data);
+      setLessonData(lessonDetails.data);
+      alert(`${lessonDetails.data[0].name} clicked`);
```

Comment:
Instead of alert, we can discuss on maybe using toast for notification popups or create our own with setTimeout with fade in/out animation. Toast would probably be simpler.

Reply:
it is just for Daniel to show where to find this code. He will build right side menu